src/client.rs

```diff
@@ -98,56 +147,65 @@ fn request(url: String, quarantine: bool) -> Boxed<Result<Response<Body>, String
 								})
 								.unwrap_or_default()
 								.to_string(),
+							true,
 							quarantine,
 						)
-						.await
-					} else {
-						match response.headers().get(header::CONTENT_ENCODING) {
-							// Content not compressed.
-							None => Ok(response),
-
-							// Content gzipped.
-							Some(hdr) => {
-								// Since we requested gzipped content, we expect
-								// to get back gzipped content. If we get
-								// back anything else, that's a problem.
-								if hdr.ne("gzip") {
-									return Err("Reddit response was encoded with an unsupported compressor".to_string());
-								}
-
-								// The body must be something that implements
-								// std::io::Read, hence the conversion to
-								// bytes::buf::Buf and then transformation into a
-								// Reader.
-								let mut decompressed: Vec<u8>;
-								{
-									let mut aggregated_body = match body::aggregate(response.body_mut()).await {
-										Ok(b) => b.reader(),
-										Err(e) => return Err(e.to_string()),
-									};
-
-									let mut decoder = match gzip::Decoder::new(&mut aggregated_body) {
-										Ok(decoder) => decoder,
-										Err(e) => return Err(e.to_string()),
-									};
-
-									decompressed = Vec::<u8>::new();
-									match io::copy(&mut decoder, &mut decompressed) {
-										Ok(_) => {}
-										Err(e) => return Err(e.to_string()),
-									};
```

Comment:
This can be simplified to checking if it's an error, since the Ok branch is empty. Though it's up to you if it's worth it to do so.

Reply:
See https://github.com/libreddit/libreddit/pull/617#discussion_r1014505284